### PR TITLE
CHE-3774 Add Eclipse Vert.x stack to default assembly

### PR DESCRIPTION
**Pull request by sbryzak**
Signed-off-by: Shane Bryzak <sbryzak@redhat.com>

This pull request adds the Eclipse Vert.x stack to the default Che assembly.

### What issues does this PR fix or reference?

CHE-3774

#### Changelog
Adds the Eclipse Vert.x stack to the default Che assembly.

#### Release note
We've added a new stack to for the Eclipse Vert.x project. This is a very popular toolkit for building reactive applications on the JVM. You can now build vert.x applications by simply creating a workspace from the vert.x stack - no configuration necessary!

### CODE REVIEW

**Review by TylerJewell — APPROVED** [review]

**Review by l0rd — APPROVED** [review]
Thank you @sbryzak for this PR! That's great to have a vertX stack. I'm wondering if we could also add the Dockerfile to https://github.com/eclipse/che-dockerfiles and double check with @dharmit if registry.centos.org/dharmit/vertx is the definitive repository or if we want to move it to something like registry.centos.org/che-stacks/vertx.

**Comment by slemeur** [comment]
Thanks @sbryzak, that's really nice to have a vert.x stack. 

I would like to suggest two things to associate to this PR:
1- We could provide to end user a sample vert.x project which help them to get started. 
2- We could maybe provide a nice tutorial about Vert.x in Che - this could maybe be referenced on Vert.x documentation too.

wdyt @sbryzak ? 
My guess is that 1 is important and 2 a nice to have ;)

**Comment by slemeur** [comment]
PR added to release note.
Added doc issue: https://github.com/eclipse/che-docs/issues/89

**Comment by sbryzak** [comment]
@l0rd I've e-mailed Dharmit to confirm which repo should be the definitive one.  @slemeur I've also asked whether a sample project and tutorial can be provided, will update back here once I get a response.

**Comment by sbryzak** [comment]
@slemeur The following example may be used, what is the best/proper way to include this in Che?

https://github.com/cescoffier/vertx-with-che

**Comment by sbryzak** [comment]
This pull request has been updated with the definitive repository registry.centos.org/che-stacks/vertx.  It should be fine to merge it now.

**Comment by benoitf** [comment]
ci-build

**Review by benoitf — APPROVED** [review]

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 1]
```diff
@@ -2064,5 +2064,76 @@
...
+    }, {
+        "name": "JDK",
+        "version": "1.8.0_45"
+    }, {
+        "name": "Maven",
+        "version": "3.2.2"
+    }],
+    "source": {
+        "type": "image",
+        "origin": "registry.centos.org/che-stacks/vertx"
```
Comment (benoitf) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json:2094]: @TylerJewell does it require an update of https://github.com/eclipse/che/blob/master/dockerfiles/cli/version/5.2.0/images-stacks

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 2]
```diff
@@ -2064,5 +2064,76 @@
...
+                            "org.eclipse.che.ssh"
+                        ],
+                        "servers": {},
+                        "attributes": {
+                            "memoryLimitBytes": "2147483648"
+                        }
+                    }
+                },
+                "recipe": {
+                    "location": "registry.centos.org/dharmit/vertx",
```
Comment (dharmit) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: @sbryzak should we replace this with `registry.centos.org/che-stacks/vertx` as well?

Comment (sbryzak) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json]: Oops, I missed that one - updated now.

File: ide/che-core-ide-stacks/src/main/resources/stacks.json (hunk 1 above)
Comment (cescoffier) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json:2094]: I don't know what's in the provided image. However, the later versions the better for Vert.x. These JDK and Maven versions are the lower bounds, any more up to date version would be fine.

Comment (dharmit) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json:2094]: > I don't know what's in the provided image.

@cescoffier: you can find the Dockerfile and other files for the image [here](https://github.com/eclipse/che-dockerfiles/tree/master/recipes/centos_vertx).

Comment (cescoffier) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json:2094]: Thanks, @dharmit. So it's Maven 3.3 (the one we used). 

Note for me: don't forget to update the docker file after Vert.x releases.

**Comment by benoitf** [comment]
@JamesDrummond ok ?

**Comment by l0rd** [comment]
@TylerJewell we added vert.x image in the CLI image-stacks lists for version [5.3.0](https://github.com/eclipse/che/pull/3843/files#diff-fa9efde6b1e97ad1f81458b1bc56e542) and [nightly](https://github.com/eclipse/che/pull/3843/files#diff-61953898574557844af7fd3e9c6f2b60). Is it good for you?

File: dockerfiles/cli/version/5.3.0/images-stacks [hunk 3]
```diff
@@ -20,3 +20,4 @@ eclipse/ubuntu_jdk8
 eclipse/ubuntu_jre
 eclipse/ubuntu_python
 eclipse/ubuntu_wildfly8
+registry.centos.org/che-stacks/vertx
```
Comment (benoitf) [inline dockerfiles/cli/version/5.3.0/images-stacks]: missing the newline

**Comment by JamesDrummond** [comment]
@slemeur Can you check IP with Eclipse foundation?

**Review by JamesDrummond — APPROVED** [review]

**Comment by benoitf** [comment]
@l0rd adding newline at end of file in dockerfiles/cli/version/5.3.0/images-stacks and I think you can merge

**Comment by l0rd** [comment]
@benoitf Thanks!

**Comment by bmicklea** [comment]
Please remember to add the correct milestone when merging PRs - it makes it easier for PM to track for changelog and release notes.

**Comment by l0rd** [comment]
@bmicklea ok will do that

**Pull request merged**